### PR TITLE
Bookmarks UI updates + Rearrange

### PR DESCRIPTION
**Pull request by joshliebe**
Task/Issue URL: https://app.asana.com/0/488551667048375/1206202020825288/f

### Description
- This adds the ability to re-order bookmarks.
- Favorites and bookmarks are no longer separated (A bookmark can now be a favorite, as indicated by the star).
- Also includes several other copy and UI changes relating to bookmarks

### Steps to test this PR

_Add bookmark_
- [ ] Go to a site and "Add Bookmark"
- [ ] Verify that the new bottom sheet is shown
- [ ] Verify that "Add to Favorites" and "Edit Bookmark" work as expected

_Import / export_
- [ ] Add some more bookmarks / favorites
- [ ] Go to Bookmarks
- [ ] Create some folders
- [ ] Tap the overflow menu and export
- [ ] Delete everything
- [ ] Verify that new empty view is shown
- [ ] Tap the overflow
- [ ] Verify that "Export" is disabled
- [ ] Tap "Import Bookmarks"
- [ ] Import the bookmarks
- [ ] Verify that the bookmarks import correctly

_Bookmarks list_
- [ ] Drag the folders and bookmarks around
- [ ] Verify that the items are reordered correctly
- [ ] Click the overflow next to an item
- [ ] Tap "Add to Favorites" / "Remove From Favorites"
- [ ] Drag the favorites around
- [ ] Delete some items
- [ ] Verify that this works as expected
- [ ] Search the list (There must be more than 5 items)
- [ ] Verify that search works as expected

_New tab_
- [ ] When you have at least one favorite, open a new tab
- [ ] Long press the favorite
- [ ] Remove the favorite
- [ ] Verify that the favorite is removed
- [ ] Add the favorite back again (In the bookmarks list)
- [ ] Go back to the new tab
- [ ] Long press the favorite
- [ ] Delete the favorite
- [ ] Verify that the bookmark is removed

_Favorites widget_
- [ ] When you have no favorites, add the widget
- [ ] Verify that the link to add favorites has been removed
- [ ] Verify that tapping the empty list opens DDG

### UI changes
See: https://app.asana.com/0/1142021229838617/1206152781445694/f

### CODE REVIEW

**Comment by joshliebe** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #4034** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4034?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4034?utm_source=stack-comment).

File: app/src/main/java/com/duckduckgo/app/systemsearch/SystemSearchViewModel.kt [hunk 1]
```diff
@@ -359,6 +366,12 @@ class SystemSearchViewModel @Inject constructor(
         }
     }
 
+    fun deleteSavedSiteSnackbarDismissed(savedSite: SavedSite) {
+        appCoroutineScope.launch(dispatchers.io()) {
+            savedSitesRepository.delete(savedSite, true)
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/systemsearch/SystemSearchViewModel.kt:371]: why do you need to delete the bookmark here? isn’t it just a favourite?

File: app/src/main/res/layout/bottom_sheet_add_bookmark.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    <FrameLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+        <com.duckduckgo.common.ui.view.listitem.OneLineListItem
+            android:id="@+id/bookmarksBottomSheetDialogPrimaryItem"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            app:leadingIconBackground="circular"
+            app:primaryText="Primary Item" />
```
Comment (malmstein) [inline app/src/main/res/layout/bottom_sheet_add_bookmark.xml:47]:
```suggestion
            tools:primaryText="Primary Item" />
```

File: app/src/main/res/layout/bottom_sheet_add_bookmark.xml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            android:layout_marginEnd="16dp" />
+    </FrameLayout>
+
+
+    <com.duckduckgo.common.ui.view.listitem.OneLineListItem
+        android:id="@+id/bookmarksBottomSheetDialogSecondaryItem"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:leadingIconBackground="circular"
+        app:primaryText="Secondary Item" />
```
Comment (malmstein) [inline app/src/main/res/layout/bottom_sheet_add_bookmark.xml:63]:
```suggestion
        tools:primaryText="Secondary Item" />
```

**Review by malmstein — APPROVED** [review]
looks great, just a few comments for organisation.

File: app/src/main/java/com/duckduckgo/app/systemsearch/SystemSearchViewModel.kt (hunk 1 above)
Comment (joshliebe) [reply app/src/main/java/com/duckduckgo/app/systemsearch/SystemSearchViewModel.kt:371]: No, that's `deleteFavoriteSnackbarDismissed` which only removes the favorite.
This needs to delete the favorite's associated bookmark.

**Pull request merged**